tools/generator/dfg/stm32/stm_peripherals.py

```diff
@@ -105,12 +105,31 @@
         'instances': '*',
         'groups': [
             {
-                'hardware': 'stm32',
+                'hardware': 'stm32-mux',
                 'features': [],
                 'protocols': ['mem2mem', 'mem2per', 'per2per'],
-                'devices': [{'family': ['f0', 'f1', 'f3']}]
-            },{
-                'hardware': 'stm32-extended',
+                'devices': [{'family': ['h7']}]
+            },
+            {
+                'hardware': 'stm32-mux',
+                'features': [],
+                'protocols': ['mem2mem', 'mem2per', 'per2per'],
+                'devices': [{'family': ['l4'], 'name': ['r5', 'r7', 'r9', 's5', 's7', 's9']}]
+            },
+            {
+                'hardware': 'stm32-stream-channel',
+                'features': [],
+                'protocols': ['mem2mem', 'mem2per', 'per2per'],
+                'devices': [{'family': ['f2', 'f4', 'f7']}]
+            },
+            {
+                'hardware': 'stm32-channel-request',
+                'features': [],
+                'protocols': ['mem2mem', 'mem2per', 'per2per'],
+                'devices': [{'family': ['l4']}]
```

Comment:
STM32F09x devices with the "STM32F091_dma_v1_1" IP also have a request mux like the L4.

Reply:
I roughly cross-checked the generated device files with the reference manuals. Almost all data looks reasonable. Only the requests for F09x devices are missing.

Reply:
Not just F091 and F098 devices use requests, but also the F030 with size C, and all L0 and L4. ST is weird.